### PR TITLE
proxy: remove connections on shutdown

### PR DESCRIPTION
**Pull request by conradludgate**
## Problem

On shutdown, proxy connections are staying open.

## Summary of changes

Remove the connections on shutdown

## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/http/conn_pool.rs [hunk 1]
```diff
@@ -64,13 +64,13 @@ pub struct EndpointConnPool {
...
-/// But probably good enough for in memory only hashes.
+/// 4096 is the number of rounds that SCRAM-SHA-256 recommends.
+/// It's not the 600,000 that OWASP recommends... but our passwords are high entropy anyway.
 ///
-/// Still takes 3.5ms to hash on my hardware.
+/// Still takes 1.4ms to hash on my hardware.
 /// We don't want to ruin the latency improvements of using the pool by making password verification take too long
 const PARAMS: Params = Params {
-    rounds: 10_000,
+    rounds: 4096,
```
Comment (conradludgate) [inline proxy/src/http/conn_pool.rs:73]: drive-by change. not related to this PR

**Comment by petuhovskiy** [comment]
Maybe let's also update `MAX_CONNS_PER_ENDPOINT` to `10`? The reasoning is that 20*6 > 100 (default max_connections).

**Comment by petuhovskiy** [comment]
Also, we'll need to kill all existing staging proxies after deploying this PR. @conradludgate can you do it? Other than that, LGTM.

**Review by petuhovskiy — APPROVED** [review]
